src/full/Agda/Interaction/Highlighting/HTML.hs

```diff
@@ -197,15 +202,20 @@ tokenStream contents info =
 
 -- | Constructs the HTML displaying the code.
 
-code :: [(Int, String, Aspects)]
+code :: Bool          -- ^ Whether to generate non-code contents as-is
+     -> [(Int, String, Aspects)]
      -> Html
-code = mconcat . map mkHtml
+code preserveLiterate = mconcat . snd . List.mapAccumL mkHtml True
   where
-  mkHtml :: (Int, String, Aspects) -> Html
-  mkHtml (pos, s, mi) =
+  mkHtml :: Bool -> (Int, String, Aspects) -> (Bool, Html)
+  mkHtml pre (pos, s, mi) = case (preserveLiterate, aspect mi) of
+    (True, Just Background) -> (pre, preEscapedToHtml s)
+    (True, Just Markup)     -> (not pre, preEscapedToHtml s')
+      where s' :: String
+            s' = if pre then "<pre class=\"agda-code\">" else "</pre>"
```

Comment:
I'm having problems with the code here.

According to the added test, we have the delimiters successfully recognized as `Markup`s. However, when I test with a simple `.lagda.md` file, the delimiters are still as-is instead of becoming paired `<pre class="agda-code">` and `</pre>`. Am I doing anything wrong here?

Reply:
You shouldn't write raw html in strings. We are using `blaze-html` to generate the code
and it has a [`pre`](https://hackage.haskell.org/package/blaze-html-0.9.1.1/docs/Text-Blaze-Html5.html#v:pre) construct to which you can attach an attribute such as [`class_`](https://hackage.haskell.org/package/blaze-html-0.9.1.1/docs/Text-Blaze-Html5-Attributes.html#v:class_) the usual way.

Reply:
That'll take a little more time to deal with, but it doesn't look like the cause of the problem to me.
I'll fix it anyway